hanjaelee/services/userService.js

```diff
@@ -16,10 +17,46 @@ const signup = async (name, email, password, profileImage) => {
   return isCreated
 }
 
-const makeHashedPassword = (password, saltRounds) => {
-  return bcrypt.hash(password, saltRounds)
+const login = async (email, password) => {
+  try {
+    let accessToken = null
+
+    const userId = await userDao.getUserId(email)
+    const hashedPassword = await userDao.getPassword(userId)
+    if (!userId || !password) return accessToken
+
+    const isSame = await checkHashedPassword(password, hashedPassword)
+    if (!isSame) return accessToken
+
+    const current = getCurrentTimeInSeconds()
+    const payLoad = {
+      iss: 'Hanjae Lee',
+      iat: current,
+      exp: current + (60 * 60 * 24),
+      userId: userId
+    }
+
+    const secretKey = process.env.SECRET_KEY
+    accessToken = jwt.sign(payLoad, secretKey)
+    return accessToken
+  } catch (err) {
+    console.error(err)
+    throw err
+  }
 }
 
+const makeHashedPassword = async (password, saltRounds) => {
+  return await bcrypt.hash(password, saltRounds)
+}
+
+const checkHashedPassword = async (password, hashedPassword) => {
+  return await bcrypt.compare(password, hashedPassword)
+}
+
+const getCurrentTimeInSeconds = () => {
+  return Math.floor((Date.now() / 1000) + (60 * 60 * 9))
```

Comment:
리턴값을 포함하는 연산을 따로 함수로 만들어서 적용해 보는 건 어떨까요?
아래는 제가 임의로 변경한 함수이니 참고해서 작성해 보세요~
```suggestion
const getCurrentDateSeconds = () => {
    return (Date.now() / 1000) + (60 * 60 * 9)
}

const dateSeconds = getCurrentDateSeconds = ()
  return Math.floor(dateSeconds)
```